drivers/kw2xrf/kw2xrf.c

```diff
@@ -0,0 +1,855 @@
+{
+#ifdef DEBUG_ENABLED
+    uint16_t addr;
+    kw2xrf_read_iregs(MKW2XDMI_MACSHORTADDRS0_LSB, (uint8_t *)&addr, sizeof(addr));
+    return addr;
+#else
+    return radio_address;
+#endif
+}
+
+uint64_t kw2xrf_set_address_long(uint64_t addr)
+{
+    radio_address_long = addr;
+
+    uint8_t buf[8];
+    buf[0] = (uint8_t)addr;
+    buf[1] = (uint8_t)(addr >> 8);
+    buf[2] = (uint8_t)(addr >> 16);
+    buf[3] = (uint8_t)(addr >> 24);
+    buf[4] = (uint8_t)(addr >> 32);
+    buf[5] = (uint8_t)(addr >> 40);
+    buf[6] = (uint8_t)(addr >> 48);
+    buf[7] = (uint8_t)(addr >> 56);
+    kw2xrf_write_iregs(MKW2XDMI_MACLONGADDRS0_0, buf, 8);
+    return addr;
+}
+
+uint64_t kw2xrf_get_address_long(void)
+{
+#ifdef DEBUG_ENABLED
+    uint64_t addr;
+    kw2xrf_read_iregs(MKW2XDMI_MACLONGADDRS0_0, (uint8_t *)&addr, sizeof(addr));
+    return addr;
+#else
+    return radio_address_long;
+#endif
+}
+
+uint16_t kw2xrf_set_pan(uint16_t pan)
+{
+    radio_pan = pan;
+
+    uint8_t buf[2];
+    buf[0] = (uint8_t)pan;
+    buf[1] = (uint8_t)(pan >> 8);
+    kw2xrf_write_iregs(MKW2XDMI_MACPANID0_LSB, buf, 2);
+    return pan;
+}
+
+uint16_t kw2xrf_get_pan(void)
+{
+#ifdef DEBUG_ENABLED
+    uint16_t pan;
+    kw2xrf_read_iregs(MKW2XDMI_MACPANID0_LSB, (uint8_t *)&pan, sizeof(pan));
+    return pan;
+#else
+    return radio_pan;
+#endif
+}
+
+
+int kw2xrf_set_channel(unsigned int ch)
+{
+    radio_channel = ch;
+
+    if (ch < 11 || ch > 26) {
+        DEBUG("Invalid channel %i set. Valid channels are 11 through 26\n", ch);
+        return -1;
+    }
+
+    /*
+     * Fc = 2405 + 5(k - 11) , k = 11,12,...,26
+     *
+     * Equation for PLL frequency, MKW2xD Reference Manual, p.255 :
+     * F = ((PLL_INT0 + 64) + (PLL_FRAC0/65536))32MHz
+     *
+     */
+    ch -= 11;
+    kw2xrf_write_dreg(MKW2XDM_PLL_INT0, MKW2XDM_PLL_INT0_VAL(pll_int_lt[ch]));
+    kw2xrf_write_dreg(MKW2XDM_PLL_FRAC0_LSB, (uint8_t)pll_frac_lt[ch]);
+    kw2xrf_write_dreg(MKW2XDM_PLL_FRAC0_MSB, (uint8_t)(pll_frac_lt[ch] >> 8));
+    return ((unsigned int) ch);
+}
+
+unsigned int kw2xrf_get_channel(void)
+{
+#ifdef DEBUG_ENABLED
+    uint8_t pll_int = kw2xrf_read_dreg(MKW2XDM_PLL_INT0);
+    uint16_t pll_frac = kw2xrf_read_dreg(MKW2XDM_PLL_FRAC0_LSB);
+    pll_frac |= ((uint16_t)kw2xrf_read_dreg(MKW2XDM_PLL_FRAC0_MSB) << 8);
+
+    for (int i = 0; i < 16; i++) {
+        if ((pll_frac_lt[i] == pll_frac) && (pll_int_lt[i] == pll_int)) {
+            return i + 11;
+        }
+    }
+
+    return 0;
+#else
+    return radio_channel;
+#endif
+}
+
+void kw2xrf_set_monitor(int mode)
+{
+    uint8_t tmp;
+    tmp = kw2xrf_read_dreg(MKW2XDM_PHY_CTRL4);
+    tmp |= MKW2XDM_PHY_CTRL4_PROMISCUOUS;
+    kw2xrf_write_dreg(MKW2XDM_PHY_CTRL4, tmp);
+}
+
+int kw2xrf_get_monitor(void)
+{
+    uint8_t tmp;
+    tmp = kw2xrf_read_dreg(MKW2XDM_PHY_CTRL4);
+    return ((tmp & MKW2XDM_PHY_CTRL4_PROMISCUOUS) != 0);
+}
+
+
+int kw2xrf_set_tx_power(int pow)
+{
+    if (pow > MKW2XDRF_OUTPUT_POWER_MAX) {
```

Comment:
The function description states values -25 to 0 while the defines range from -35 to 8?